oio/cli/admin/service_list.py

```diff
@@ -0,0 +1,178 @@
+
+    def translate_cid(self, cid):
+        """Resolve a CID into account/container names."""
+        try:
+            if cid not in self._cids_cache:
+                md = self.dir.show(cid=cid)
+                self._cids_cache[cid] = '/'.join([md.get('account'),
+                                                  md.get('name')])
+            return self._cids_cache[cid]
+        except ClientException:
+            pass
+        return cid
+
+
+class RawxListContainer(ServiceListCommand):
+    """
+    Get the names of all containers which some chunks are stored in the
+    specified rawx
+    """
+    reqid_prefix = 'ACLI-RLC'
+
+    def _list_containers(self, rawx):
+        info = self.rdir.status(rawx)
+        containers = info.get('container')
+        result = {self.translate_cid(cid=k): v['total']
+                  for k, v in containers.iteritems()}
+        result['Total'] = info.get('chunk')['total']
+        return result.iteritems()
+
+    def get_parser(self, prog_name):
+        parser = super(RawxListContainer, self).get_parser(prog_name)
+        parser.add_argument(
+            'rawx_id',
+            metavar='<rawx_id>',
+            help='ID of the rawx service',
+        )
+        parser.add_argument(
+            '--limit',
+            metavar='<limit>',
+            type=int,
+            default=1000,
+            help='Limit the number of objects returned (1000 by default)'
+        )
+        return parser
+
+    def take_action(self, parsed_args):
+        super(RawxListContainer, self).take_action(parsed_args)
+        return ('Name', 'Chunks'), self._list_containers(parsed_args.rawx_id)
+
+
+class Meta2ListContainer(ServiceListCommand):
+    """
+    Get the names of all containers which are in the specified meta2
+    """
+    reqid_prefix = 'ACLI-M2LC'
+
+    def get_parser(self, prog_name):
+        from oio.cli.common.utils import ValueFormatStoreTrueAction
+        parser = super(Meta2ListContainer, self).get_parser(prog_name)
+        parser.add_argument(
+            'meta2_id',
+            metavar='<meta2_id>',
+            help='ID of the meta2 service'
+        )
+        parser.add_argument(
+            '--no-paging',
+            dest='no_paging',
+            default=False,
+            help=("List all elements without paging "
+                  "(and set output format to 'value')"),
+            action=ValueFormatStoreTrueAction,
+        )
+        parser.add_argument(
+            '--prefix',
+            metavar='<prefix>',
+            help='Filter list using <prefix>'
+        )
+        parser.add_argument(
+            '--marker',
+            metavar='<marker>',
+            help='Marker for paging'
+        )
+        parser.add_argument(
+            '--limit',
+            metavar='<limit>',
+            type=int,
+            default=1000,
+            help='Limit the number of objects returned (1000 by default)'
+        )
```

Comment:
Service ID, `--limit`, `--marker`, `--no-paging` and `--prefix` can apply to both `meta2 list` and `rawx list` commands, thus they can be implemented in `ServiceListCommand` class.

Also, the help string for `--marker` should tell that this marker is not one of the items returned, but a specific marker printed by the tool, aside from the list.

Reply:
Ok, for the first version of the tool, we chose not to implement them for rawx operations.